### PR TITLE
Improve cost of BLS key serialization for gRPC

### PR DESCRIPTION
**Pull request by hexfusion**
# Note: this will require a protocol version bump 

## Why this should be merged

This PR is a followup to https://github.com/ava-labs/avalanchego/pull/1326 where we benchmarked the serialzation costs of BLS public keys. The profile from bench showed that the majority of the cost was spent in CGO on very expensive operations including Compressing, Decompressing bytes and re-verifying the keys. Verification was not explicitly required because the keys were already verified before being added to the P-Chain.

### before 
```
$ go test -benchmem -run=^$ -bench ^BenchmarkGetValidatorSet$ github.com/ava-labs/avalanchego/snow/validators/gvalidators -memprofile benchvset.mem -cpuprofile benchvset.cpu

goos: linux
goarch: amd64
pkg: github.com/ava-labs/avalanchego/snow/validators/gvalidators
cpu: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
BenchmarkGetValidatorSet
BenchmarkGetValidatorSet/get_validator_set_1_validators
BenchmarkGetValidatorSet/get_validator_set_1_validators-12                  2175            534900 ns/op           10057 B/op        190 allocs/op
BenchmarkGetValidatorSet/get_validator_set_16_validators
BenchmarkGetValidatorSet/get_validator_set_16_validators-12                  350           3139479 ns/op           21040 B/op        300 allocs/op
BenchmarkGetValidatorSet/get_validator_set_32_validators
BenchmarkGetValidatorSet/get_validator_set_32_validators-12                  204           5763480 ns/op           33196 B/op        414 allocs/op
BenchmarkGetValidatorSet/get_validator_set_1024_validators
BenchmarkGetValidatorSet/get_validator_set_1024_validators-12                  7         156047375 ns/op          794016 B/op       7437 allocs/op
BenchmarkGetValidatorSet/get_validator_set_2048_validators
BenchmarkGetValidatorSet/get_validator_set_2048_validators-12                  4         310664506 ns/op         1607208 B/op      14673 allocs/op
```

### after 

```
$ go test -benchmem -run=^$ -bench ^BenchmarkGetValidatorSet$ github.com/ava-labs/avalanchego/snow/validators/gvalidators -memprofile benchvset.mem -cpuprofile benchvset.cpu

goos: linux
goarch: amd64
pkg: github.com/ava-labs/avalanchego/snow/validators/gvalidators
cpu: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
BenchmarkGetValidatorSet
BenchmarkGetValidatorSet/get_validator_set_1_validators
BenchmarkGetValidatorSet/get_validator_set_1_validators-12                  4336            264489 ns/op           10204 B/op        190 allocs/op
BenchmarkGetValidatorSet/get_validator_set_16_validators
BenchmarkGetValidatorSet/get_validator_set_16_validators-12                 2874            385106 ns/op           23575 B/op        299 allocs/op
BenchmarkGetValidatorSet/get_validator_set_32_validators
BenchmarkGetValidatorSet/get_validator_set_32_validators-12                 2376            494729 ns/op           38050 B/op        412 allocs/op
BenchmarkGetValidatorSet/get_validator_set_1024_validators
BenchmarkGetValidatorSet/get_validator_set_1024_validators-12                249           4549176 ns/op          959468 B/op       7385 allocs/op
BenchmarkGetValidatorSet/get_validator_set_2048_validators
BenchmarkGetValidatorSet/get_validator_set_2048_validators-12                142           8424447 ns/op         1917264 B/op      14573 allocs/op
```
### profiles
[benchvset.cpu.txt](https://github.com/ava-labs/avalanchego/files/11225953/benchvset.cpu.txt)
[benchvset.mem.txt](https://github.com/ava-labs/avalanchego/files/11225959/benchvset.mem.txt)

Closes https://github.com/ava-labs/avalanchego/issues/1327

## How this works

Remove and replace unnecessarily expensive calls to CGO with less expensive variants.

## How this was tested

manually

### CODE REVIEW

File: vms/platformvm/signer/proof_of_possession.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func NewProofOfPossession(sk *bls.SecretKey) *ProofOfPossession {
 }
 
 func (p *ProofOfPossession) Verify() error {
-	publicKey, err := bls.PublicKeyFromBytes(p.PublicKey[:])
+	publicKey, err := bls.DecompressPublicKey(p.PublicKey[:])
```
Comment (StephenButtolph) [inline vms/platformvm/signer/proof_of_possession.go:46]: This no longer correctly validates the public key - which would open up a terrifying series of vulns

Comment (hexfusion) [reply vms/platformvm/signer/proof_of_possession.go:46]: Will verify all paths and add some tests since the core looks reasonable but yeah..

**Review by aaronbuchwald — APPROVED** [review]

**Review by abi87 — APPROVED** [review]
lgtm, there may be other places in the codebase where we can use this optimization (matter for a different PR)

**Comment by hexfusion** [comment]
> lgtm, there may be other places in the codebase where we can use this optimization (matter for a different PR)

Yeah I was going to break out the logic into funcs my fear is someone using the it in the wrong place (like me :)) to grab a perf gain and opening vuln.

**Pull request merged**